gs_quant/api/fred/fredapi.py

```diff
@@ -0,0 +1,305 @@
+            * *search_critera* (``str``) --
+                'text', 'release_id', or 'category_id'
+        :param order_by: Order the results by a criterion ('search_rank', 'field', 'title', 'units', 'frequency',
+            'seasonal_adjustment', 'realtime_start', 'realtime_end', 'last_updated', 'start', 'end',
+            'popularity')
+        :param filter: Filter the results by a criterion ('frequency', 'units', and 'seasonal_adjustment')
+        :return: A DataFrame of the Fred series covered
+
+        **Examples**
+
+        >>> from gs_quant.api.fred import FredDataApi
+        >>> 
+        >>> API_KEY = <YOUR API KEY>
+        >>> fredAPI = FredDataApi(api_key = API_KEY)
+        >>> fred_data = Dataset("FRED", fredAPI)
+
+        >>> fred_data.get_coverage(search_item="GDP", search_criteria="text")
+        >>> fred_data.get_coverage(search_item=151, search_criteria="release_id")
+        >>> fred_data.get_coverage(search_item=33058, search_criteria="category_id")
+        """
+        
+        assert("search_criteria" in kwargs)
+        assert("search_item" in kwargs)
+        if limit is None:
+            limit = 1000
+        search_critera = kwargs["search_criteria"]
+        search_item = kwargs["search_item"]
+
+        if search_critera == 'text':
+            url = "{}/series/search?search_text={}&limit=1000".format(self.root_url, requote_uri(search_item))
+            coverage = self.__get_search_results(url=url, limit=limit)
+            if coverage is None:
+                raise ValueError('No series exists for text: ' + str(search_item))
+        elif search_critera == 'release_id':
+            url = "{}/release/series?release_id={}&limt=1000".format(self.root_url, search_item)
+            coverage = self.__get_search_results(url=url, limit=limit)
+            if coverage is None:
+                raise ValueError('No series exists for release id: ' + str(search_item))
+        elif search_critera == 'category_id':
+            url = "{}/category/series?category_id={}&limit=1000".format(self.root_url, search_item)
+            coverage = self.__get_search_results(url=url, limit=limit)
+            if coverage is None:
+                raise ValueError('No series exists for category id: ' + str(search_item))
+        else:
+            raise ValueError("Please specify search_critera as: 'text', 'release_id', or 'category'!")
+
+        return coverage
+
+    def construct_dataframe_with_types(
+        self, 
+        dataset_id: str, 
+        data: dict
+    ) -> pd.DataFrame:
+        """
+        Constructs a dataframe with correct date types.
+        
+        :param data: Data to convert with correct types
+        :return: Dataframe with correct types
+        """
+        if len(data) > 0:
+            idx = pd.date_range(dt.date(1776, 7, 4), dt.date(2262, 4, 11))
+            df = None
+            for series in data.values():
+                if df is None:
```

Comment:
isn't this always going to be true given assignment in the prev line?

Reply:
not necessarily since after the first iteration of the loop, df won't be None anymore